src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -3667,6 +3682,72 @@ int wm_vuldet_json_parser(char *json_path, wm_vuldet_db *parsed_vulnerabilities,
 }
 
 
+void wm_insert_rpm_source_package(sqlite3 *db, agent_software *agent, const char *vendor, const char *product, const char *source,
+                                    const char *version, const char *arch) {
+    char *tok = NULL;
+    char *src_name = NULL;
+    char *src_version = NULL;
+    char *saved = NULL;
+    char *tmp_src = NULL;
+    size_t len = 0;
+
+    /* Discard all sources which have same name as
+     their respective binary. */
+    if (strstr(source, product)) {
```

Comment:
Shouldn't we use `strcmp`? Or aren't we looking for exactly the same name?

Reply:
Unfortunately that's an invalid comparison, since the source comes as `name-version-release.src.rpm ` and product just as `name`. The only viable solution is to check wether product is contained within source; that is, the binary's name will always contain the source's name. For instance:
source -> `mysql-4.5.2-1.src.rpm` 
binary -> `mysql-server-4.5.2.1.rpm`
product -> `mysql-server`

Since product is not contained within source, we can guess they are diferrent names.

Reply:
Good!